metadata-ingestion/src/datahub/ingestion/source/dbt/dbt_common.py

```diff
@@ -707,6 +850,182 @@ def filter_nodes(self, all_nodes: List[DBTNode]) -> List[DBTNode]:
 
         return nodes
 
+    @staticmethod
+    def _to_schema_info(schema_fields: List[SchemaField]) -> SchemaInfo:
+        return {column.fieldPath: column.nativeDataType for column in schema_fields}
+
+    def _infer_schemas_and_update_cll(self, all_nodes_map: Dict[str, DBTNode]) -> None:
+        if not self.config.infer_dbt_schemas:
+            if self.config.include_column_lineage:
+                raise ConfigurationError(
+                    "`infer_dbt_schemas` must be enabled to use `include_column_lineage`"
+                )
+            return
+
+        graph = self.ctx.graph
+
+        schema_resolver = SchemaResolver(
+            platform=self.config.target_platform,
+            platform_instance=self.config.target_platform_instance,
+            env=self.config.env,
+        )
+
+        target_platform_urn_to_dbt_name: Dict[str, str] = {}
+
+        # Iterate over the dbt nodes in topological order.
+        # This ensures that we process upstream nodes before downstream nodes.
+        for dbt_name in topological_sort(
+            list(all_nodes_map.keys()),
+            edges=list(
+                itertools.chain.from_iterable(
+                    [(upstream, node.dbt_name) for upstream in node.upstream_nodes]
+                    for node in all_nodes_map.values()
+                )
+            ),
```

Comment:
```suggestion
            edges=[(upstream, node.dbt_name) for node in all_nodes_map.values() for upstream in node.upstream_nodes],
```
Pretty sure you can just do this